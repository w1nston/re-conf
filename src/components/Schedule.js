import React, { Component } from 'react';
import SwipeableView from 'react-swipeable-views';
import styled from 'styled-components';
import Day from './Day';

export const Title = styled.h1`
  font-family: 'Love Ya Like A Sister', 'Arial', 'sans-serif';
  font-size: 34px;
`;

export default class Schedule extends Component {
  constructor(props) {
    super(props);
    this.days = props.schedule ? Object.keys(props.schedule) : [];
    this.state = {
      title: this.days[0],
    };
  }

  changeTitle = newIndex =>
    this.setState(() => ({ title: this.days[newIndex] }));

  render() {
    return (
      <section>
        <Title>May {this.state.title}</Title>
        <SwipeableView onChangeIndex={this.changeTitle}>
          {this.days.map(day => (
            <Day key={day} items={this.props.schedule[day]} />
          ))}
        </SwipeableView>
      </section>
    );
  }
}
