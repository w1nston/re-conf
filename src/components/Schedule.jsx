import React, { Component } from 'react';
import SwipeableViews from 'react-swipeable-views';
import Day from './Day';

const DAY = {
  0: 'May 16',
  1: 'May 17',
  2: 'May 18',
  3: 'May 19',
  4: 'May 20',
};

export default class Schedule extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: DAY[0],
    };
  }

  changeTitle = index => this.setState(() => ({ title: DAY[index] }));

  render() {
    const { schedule = {} } = this.props;
    return (
      <section>
        <h1>{this.state.title}</h1>
        <SwipeableViews onChangeIndex={this.changeTitle}>
          <Day events={schedule[16]} />
          <Day events={schedule[17]} />
          <Day events={schedule[18]} />
          <Day events={schedule[19]} />
          <Day events={schedule[20]} />
        </SwipeableViews>
      </section>
    );
  }
}
