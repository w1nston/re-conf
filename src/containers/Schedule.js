import { compose, mapProps } from 'recompose';
import addHours from 'date-fns/add_hours';
import parse from 'date-fns/parse';
import getDate from 'date-fns/get_date';
import compareAsc from 'date-fns/compare_asc';
import Schedule from '../components/Schedule';

const extractEvents = props => {
  const { data, ...rest } = props;
  const { events } = data;
  return {
    events,
    ...rest,
  };
};

const extractScheduleObject = props => {
  const { events, ...rest } = props;
  const [schedule] = events;
  return {
    schedule,
    ...rest,
  };
};

const extractSchedule = props => {
  const { schedule } = props.schedule;
  return {
    ...props,
    schedule,
  };
};

const transformSchedule = props => {
  const weekSchedule = props.schedule.reduce((acc, event) => {
    const date = addHours(parse(event.startDate), 2);
    const day = getDate(date);
    const e = {
      ...event,
      startDate: date,
    };

    if (acc[day]) {
      acc[day].push(e);
    } else {
      acc[day] = [e];
    }

    return acc;
  }, {});

  return {
    ...props,
    schedule: weekSchedule,
  };
};

const sortSchedule = props => {
  const ascending = (x, y) => compareAsc(x.startDate, y.startDate);
  const { schedule } = props;
  Object.keys(schedule).forEach(day => schedule[day].sort(ascending));
  return {
    ...props,
    schedule,
  };
};

export default compose(
  mapProps(extractEvents),
  mapProps(extractScheduleObject),
  mapProps(extractSchedule),
  mapProps(transformSchedule),
  mapProps(sortSchedule)
)(Schedule);
