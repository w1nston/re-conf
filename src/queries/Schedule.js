import { branch, compose, renderComponent } from 'recompose';
import { gql, graphql } from 'react-apollo';
import Schedule from '../containers/Schedule';
import Loading from '../components/Loading';

const slug = 'reacteurope-2017';

const withLoading = branch(
  props => props.data.loading,
  renderComponent(Loading)
);

const query = gql`
  query conferenceSchedule {
    events(slug: "${slug}") {
      schedule {
        id,
        title
        description
        startDate
        type
        length
        speakers {
          id,
          name
          avatarUrl
          twitter,
        }
      }
    }
  }
`;

export default compose(graphql(query), withLoading)(Schedule);
