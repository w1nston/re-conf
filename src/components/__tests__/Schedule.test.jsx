import React from 'react';
import { shallow } from 'enzyme';
import Schedule from '../Schedule';

describe('<Schedule/>', () => {
  it('renders', () => {
    const component = shallow(<Schedule />);
    expect(component).toMatchSnapshot();
  });
});
