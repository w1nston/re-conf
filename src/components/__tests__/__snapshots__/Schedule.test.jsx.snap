exports[`<Schedule/> renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section>
    <h1>
        May 16
    </h1>
    <ReactSwipableView
        animateHeight={false}
        animateTransitions={true}
        axis="x"
        disabled={false}
        enableMouseEvents={false}
        hysteresis={0.6}
        ignoreNativeScroll={false}
        index={0}
        onChangeIndex={[Function]}
        resistance={false}
        springConfig={
            Object {
                "delay": "0s",
                "duration": "0.35s",
                "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
              }
        }
        threshold={5}>
        <Day
            events={undefined} />
        <Day
            events={undefined} />
        <Day
            events={undefined} />
        <Day
            events={undefined} />
        <Day
            events={undefined} />
    </ReactSwipableView>
</section>,
  "nodes": Array [
    <section>
      <h1>
            May 16
      </h1>
      <ReactSwipableView
            animateHeight={false}
            animateTransitions={true}
            axis="x"
            disabled={false}
            enableMouseEvents={false}
            hysteresis={0.6}
            ignoreNativeScroll={false}
            index={0}
            onChangeIndex={[Function]}
            resistance={false}
            springConfig={
                  Object {
                        "delay": "0s",
                        "duration": "0.35s",
                        "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                      }
            }
            threshold={5}>
            <Day
                  events={undefined} />
            <Day
                  events={undefined} />
            <Day
                  events={undefined} />
            <Day
                  events={undefined} />
            <Day
                  events={undefined} />
      </ReactSwipableView>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Schedule />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Schedule {
        "_reactInternalInstance": [Circular],
        "changeTitle": [Function],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "title": "May 16",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section>
          <h1>
                    May 16
          </h1>
          <ReactSwipableView
                    animateHeight={false}
                    animateTransitions={true}
                    axis="x"
                    disabled={false}
                    enableMouseEvents={false}
                    hysteresis={0.6}
                    ignoreNativeScroll={false}
                    index={0}
                    onChangeIndex={[Function]}
                    resistance={false}
                    springConfig={
                              Object {
                                        "delay": "0s",
                                        "duration": "0.35s",
                                        "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                                      }
                    }
                    threshold={5}>
                    <Day
                              events={undefined} />
                    <Day
                              events={undefined} />
                    <Day
                              events={undefined} />
                    <Day
                              events={undefined} />
                    <Day
                              events={undefined} />
          </ReactSwipableView>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section>
          <h1>
                    May 16
          </h1>
          <ReactSwipableView
                    animateHeight={false}
                    animateTransitions={true}
                    axis="x"
                    disabled={false}
                    enableMouseEvents={false}
                    hysteresis={0.6}
                    ignoreNativeScroll={false}
                    index={0}
                    onChangeIndex={[Function]}
                    resistance={false}
                    springConfig={
                              Object {
                                        "delay": "0s",
                                        "duration": "0.35s",
                                        "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                                      }
                    }
                    threshold={5}>
                    <Day
                              events={undefined} />
                    <Day
                              events={undefined} />
                    <Day
                              events={undefined} />
                    <Day
                              events={undefined} />
                    <Day
                              events={undefined} />
          </ReactSwipableView>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Schedule />,
}
`;
